jetbrains-rider/ReSharper.AWS/src/AWS.Daemon/RunMarkers/LambdaRunMarkerProvider.cs

```diff
@@ -0,0 +1,371 @@
+using System.Collections.Generic;
+using System.Linq;
+using JetBrains.Annotations;
+using JetBrains.Application.Settings;
+using JetBrains.Metadata.Reader.API;
+using JetBrains.Metadata.Reader.Impl;
+using JetBrains.ProjectModel;
+using JetBrains.ProjectModel.ProjectsHost.Dependencies;
+using JetBrains.ReSharper.Feature.Services.Daemon;
+using JetBrains.ReSharper.Host.Features.RunMarkers;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.Caches.SymbolCache;
+using JetBrains.ReSharper.Psi.CSharp;
+using JetBrains.ReSharper.Psi.CSharp.Tree;
+using JetBrains.ReSharper.Psi.Resolve;
+using JetBrains.ReSharper.Psi.Tree;
+using JetBrains.ReSharper.Psi.Util;
+using JetBrains.Util;
+using JetBrains.Util.Logging;
+using IMethodDeclaration = JetBrains.ReSharper.Psi.CSharp.Tree.IMethodDeclaration;
+
+namespace ReSharper.AWS.RunMarkers
+{
+    [Language(typeof(CSharpLanguage))]
+    public class LambdaRunMarkerProvider : IRunMarkerProvider
+    {
+        private ILogger myLogger = Logger.GetLogger<LambdaRunMarkerProvider>();
+
+        private static readonly IClrTypeName LambdaContextTypeName =
+            new ClrTypeName("Amazon.Lambda.Core.ILambdaContext");
+
+        private static readonly List<string> AmazonLambdaEventNamespaces =
```

Comment:
This worries me because we have to keep it up to date, and we aren't told when new ones get added. @kiiadi 	Thoughts?

Reply:
Ok. I've slightly updated the logic to get all namespaces inside `Amazon.Lambda` with `Events` suffix and check for all classes inside those namespaces. Please let me now if this solution works better.

Reply:
Cool, that solution makes me feel better

Reply:
@kiiadi, I've added some info to the `IsSuitableLambdaMethod` method with steps to define Amazon Lambda function. Check for Amazon events parameters and return types is used to exit earlier. Logic to check for Custom Data type is a bit more complicated and less straightforward. So, first we check for any predefined Amazon events and if it matches - then we think it is lambda. Both checks for Amazon events and Custom Data requires ILambdaSerializer presence on method or assembly level.

Please note: this logic is based on AWS Lambda Handler documentation and might miss some cases. I'll appreciate if you could add some comments about steps in this pease of logic or provide some real-cases with lambda definition.